component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPUtils.java

```diff
@@ -0,0 +1,561 @@
+        try {
+            int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+            RealmService realmService = IdentityTenantUtil.getRealmService();
+            userRealm = realmService.getTenantUserRealm(tenantId);
+        } catch (Exception e) {
+            throw new AuthenticationFailedException("Cannot find the user realm for the tenant domain "
+                    + tenantDomain, e);
+        }
+        return userRealm;
+    }
+
+    /**
+     * Get the mobile number for Username.
+     *
+     * @param username the username
+     * @return mobile number
+     * @throws VoiceOTPException
+     */
+    public static String getMobileNumberForUsername(String username) throws VoiceOTPException,
+            AuthenticationFailedException {
+
+        UserRealm userRealm;
+        String mobile;
+        try {
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+            userRealm = getUserRealm(tenantDomain);
+            if (userRealm != null) {
+                mobile = userRealm.getUserStoreManager()
+                        .getUserClaimValue(tenantAwareUsername, VoiceOTPConstants.MOBILE_CLAIM, null);
+            } else {
+                throw new VoiceOTPException("Cannot find the user realm for the given tenant domain : " + tenantDomain);
+            }
+        } catch (UserStoreException e) {
+            throw new VoiceOTPException("Cannot find the user " + username + " to get the mobile number ", e);
+        }
+        return mobile;
+    }
+
+    /**
+     * Check whether VoiceOTP is mandatory or not.
+     *
+     * @param context the AuthenticationContext
+     * @return true or false
+     */
+    public static boolean isVoiceOTPMandatory(AuthenticationContext context) {
+
+        return Boolean.parseBoolean(getConfiguration(context, VoiceOTPConstants.IS_VOICEOTP_MANDATORY));
+    }
+
+    /**
+     * Check whether admin enable to send otp directly to mobile number or not.
+     *
+     * @param context the AuthenticationContext
+     * @return true or false
+     */
+    public static boolean isSendOTPDirectlyToMobile(AuthenticationContext context) {
+
+        return Boolean.parseBoolean(getConfiguration(context, VoiceOTPConstants.IS_SEND_OTP_DIRECTLY_TO_MOBILE));
+    }
+
+    /**
+     * Check whether admin enable to send otp directly to mobile number which gets from federated idp claims.
+     *
+     * @param context the AuthenticationContext
+     * @return true or false
+     */
+    public static boolean sendOtpToFederatedMobile(AuthenticationContext context) {
+
+        return Boolean.parseBoolean(getConfiguration(context, VoiceOTPConstants.IS_SEND_OTP_TO_FEDERATED_MOBILE));
+    }
+
+    /**
+     * Check whether user enable the second factor or not.
+     *
+     * @param context the AuthenticationContext
+     * @return true or false
+     */
```

Comment:
```suggestion
    /**
     * Check whether user has enabled voice OTP or not.
     *
     * @param context Authentication context.
     * @return true if VoiceOTP is enabled by user.
     */
```

Let's improve the method comments in other places as well.